Islandora-CLAW/Syn
# Add attribute to allow anonymous GET requests through Syn

Partially resolves: https://github.com/Islandora-CLAW/CLAW/issues/573

# What does this Pull Request do?

Adds another attribute to the `site` element `anonymous="(true|false)"`, to allow you to set some or all sites to allow unauthenticated GET requests.

# What's new?
This does not clean up the XML validation issues I expressed [here](https://github.com/Islandora-CLAW/CLAW/issues/573#issuecomment-298145175), but I figured those could be dealt with later.

This adds in the third parsing of the XML files to generate a map of URL and anonymous check, if a site URL matches the host that is the option used, or if the default is set and no site is specified that is what is used.

Sets credentials as `anonymous` with roles of `anonymous` and `islandora`. 

Not sure if there is a way to avoid authentication once you have required it, like to avoid creating sessions like @jonathangreen was suggesting. That can come later.

# How should this be tested?

1. Pull this branch into your claw_vagrant
2. Run `./gradlew clean build`
3. Copy the `build/lib/islandora-syn-0.0.1-all.jar` to `/var/lib/tomcat8/lib`
4. Edit your `/var/lib/tomcat8/conf/syn-settings.xml` sites file include an `anonymous='true'` option.
5. Restart tomcat

You should be able to access `http://localhost:8080/fcrepo/rest` without a token, but a POST request will still fail.

# Interested parties
@jonathangreen @ajs6f @dannylamb
